MinimalMVVM/ViewModels/Presenter.cs

```diff
@@ -1,47 +1,116 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Windows.Input;
 using MinimalMVVM.Models;
 
 namespace MinimalMVVM.ViewModels
 {
-    public class Presenter : ObservableObject
+    /*
+     * HOW IT WORKS:
+     * 1. When you type into the input TextBox, you're actually setting the Presenter class's SomeText property (via data binding). Watch the debug console to see the PropertyChange events firing when you type
+     * 2. When you click the "Convert" button (or press enter) you are running the Command specified in the convert button's Command property.
+     *      - This is data-bound to Presenter's ConvertTextCommand, which returns an instance of DelegateCommand (see relevant region below)
+     *      - So when you click the button or press enter, you are running ConvertTextCommand.Execute()
+     *      - ConvertTextCommand.Execute() ends up pointing to Presenter.ConvertText()
+     *          - When Presenter.ConvertText() is called, it shoves Presenter.SomeText through Presenter._textConverter.ConvertText(), which returns the converted all-caps string
+     *          - Then Presenter.ConvertText() adds the converted string to Presenter._history, which is data bound (via the Presenter.History property) to the history ListBox in the XAML UI
+    */ 
+
+    // The contents of Presenter are best understood when read from top to bottom!
+    // It's easy to miss, but the actual instance of Presenter that we use is created in XAML, not code-behind, in ConvertWindow.xaml
+    public class Presenter : ObservableObject // Base on the ObservableObject class so simple properties will play nice with data binding (see ObservableObject.cs)
     {
-        private readonly TextConverter _textConverter = new TextConverter(s => s.ToUpper());
-        private string _someText;
-        private readonly ObservableCollection<string> _history = new ObservableCollection<string>();
+        #region Private TextConverter Instance
+        // Create an instance of TextConverter, using an anonymous method as a parameter to pass in the conversion function since it's so simple
+        private readonly TextConverter _textConverter = new TextConverter(
+            delegate(string s)
```

Comment:
I don't like this change. You should get used to the lambda form as it is more concise. If you need to add further content you can always do:

```c#
private readonly TextConverter _textConverter = new TextConverter(s => {
    // Uncomment to see when this runs
    //Console.WriteLine("Anonymous method within TextConverter instance ran!");

    return s.ToUpper();
});
```

But it would probably be better to put the debug in the `ToUpper` method.